src/assignmentNotification.js

```diff
@@ -19,47 +19,39 @@ AssignmentNotification.prototype.send = function() {
   // the assignment notification from the analytics write success we can
   // bring this down to 1 HTTP request
 
-  var firstPersist = this._persistAssignment();
+  const firstPersist = this._persistAssignment();
 
-  var secondPersist = $.Deferred();
-  this._visitor.analytics.trackAssignment(
-    this._visitor.getId(),
-    this._assignment,
-    function(success) {
-      var promise = this._persistAssignment(success ? 'success' : 'failure');
-      promise.always(function() {
-        if (promise.state() === 'resolved') {
-          secondPersist.resolve();
-        } else {
-          secondPersist.reject();
-        }
-      });
-    }.bind(this)
-  );
+  const secondPersist = new Promise((resolve, reject) => {
+    this._visitor.analytics.trackAssignment(this._visitor.getId(), this._assignment, success =>
+      this._persistAssignment(success ? 'success' : 'failure')
+        .then(resolve)
+        .catch(reject)
+    );
+  });
 
-  return $.when(firstPersist, secondPersist);
+  return Promise.all([firstPersist, secondPersist]);
 };
 
 AssignmentNotification.prototype._persistAssignment = function(trackResult) {
-  return $.ajax(TestTrackConfig.getUrl() + '/api/v1/assignment_event', {
-    method: 'POST',
-    dataType: 'json',
-    crossDomain: true,
-    data: {
-      visitor_id: this._visitor.getId(),
-      split_name: this._assignment.getSplitName(),
-      context: this._assignment.getContext(),
-      mixpanel_result: trackResult
-    }
-  }).fail(
-    function(jqXHR, textStatus, errorThrown) {
-      var status = jqXHR && jqXHR.status,
-        responseText = jqXHR && jqXHR.responseText;
+  return client
+    .post(
+      TestTrackConfig.getUrl() + '/api/v1/assignment_event',
```

Comment:
Can we create the default axios client with `TestTrackConfig.getUrl()` as the `baseUrl`? https://www.npmjs.com/package/axios#custom-instance-defaults